src/main/java/org/interledger/Condition.java

```diff
@@ -0,0 +1,66 @@
+package org.interledger;
+
+import java.util.Arrays;
+
+/**
+ * The execution condition attached to all transfers in an Interledger payment.
+ */
+public class Condition {
+
+  private byte[] hash;
+
+  /**
+   * Create a new {@code Condition} from a SHA-256 hash.
+   * 
+   * @param hash The SHA-256 hash of a preimage that is used as the fulfillment of this condition
+   */
+  public Condition(byte[] hash) {
```

Comment:
Yeh, basically there was a decision to only support SHA-256 conditions so I want to keep this simple. We event restrict the size of fulfillments to 32 bytes.